### DISCUSSION TITLE
vips 8.15 unable to load multi-tiled dicom

### DISCUSSION THREAD

**Discussion by mollyclaretechcyte** [libvips/libvips#4017] [Q&A]
I am running `ubuntu 24.04` and installed libvips-tools via apt which gave me libvips version 8.15.1 and it came with openslide and I expect it to come with libdicom support. 

Running `vips crop my_dicom.dcm crop.png 0 0 257 256`  where my_dicom.dcm is a multi-tiled dicom that has tiles sized 256x256 pixels. Since I'm asking for a width one pixel past the first tile, the crop fails with error: `extract_area: bad extract area`. If I ask for crops 256x256 or smaller, it works. 

Do I need to install anything else, or is this expected to work?
Here is my vips-config if that's useful
```
# vips --vips-config
enable debug: false
enable deprecated: true
enable modules: true
enable cplusplus: true
enable RAD load/save: true
enable Analyze7 load/save: true
enable PPM load/save: true
enable GIF load: true
use fftw for FFTs: true
SIMD support with highway: true
accelerate loops with ORC: false
ICC profile support with lcms: true
zlib: true
text rendering with pangocairo: true
font file support with fontconfig: true
EXIF metadata support with libexif: true
JPEG load/save with libjpeg: true
JXL load/save with libjxl: true (dynamic module: true)
JPEG2000 load/save with OpenJPEG: true
PNG load/save with libspng: true
PNG load/save with libpng: false
selected quantisation package: imagequant
TIFF load/save with libtiff: true
image pyramid save with libarchive: true
HEIC/AVIF load/save with libheif: true (dynamic module: true)
WebP load/save with libwebp: true
PDF load with PDFium: false
PDF load with poppler-glib: true (dynamic module: true)
SVG load with librsvg: true
EXR load with OpenEXR: true
OpenSlide load: true (dynamic module: true)
Matlab load with libmatio: true
NIfTI load/save with niftiio: false
FITS load/save with cfitsio: true
GIF save with cgif: true
selected Magick package: MagickCore (dynamic module: true)
Magick API version: magick6
Magick load: true
Magick save: true
```

**Comment by jcupitt** [comment]
Hello @mollyclaretechcyte, could you share a complete sample image that shows this issue?

**Reply by mollyclaretechcyte** [reply]
Yes, I was able to reproduce it with [the test dicom file](https://github.com/libvips/libvips/blob/master/test/test-suite/images/dicom_test_image.dcm) that is in the libvips test suite. This is the command I execute to reproduce: `vips crop dicom_test_image.dcm cropped.png 0 0 129 128`

**Reply by jcupitt** [reply]
You can see what's happening with:

```
$ vipsheader dicom_test_image.dcm 
dicom_test_image.dcm: 128x128 ushort, 1 band, grey16, magickload
```

libvips is using imagemagick to load this dicom, so any problems are theirs, unfortunately.

You can try forcing openslide:

```
$ vips openslideload dicom_test_image.dcm x.v
openslide2vips: unsupported slide format
```

So it's not a virtual slide image, according to openslide.

**Reply by mollyclaretechcyte** [reply]
Did I choose a bad example file in this case? I expected this test file to work since its in the vips test suite.

**Reply by jcupitt** [reply]
I think that file is used to test the imagemagick loader.

openslide + libdicom will only load WG26 WSI DICOMs, it's not a general purpose DICOM loader.
